lib/vmdb/loggers.rb

```diff
@@ -83,18 +84,12 @@ def self.create_logger(log_file_name, logger_class = ManageIQ::Loggers::Base)
       File.basename(log_file_name, ".*")
     end
 
-    private_class_method def self.create_wrapper_logger(log_file_name, logger_class, wrapped_logger)
-      log_file_name = Pathname.new(log_file_name) unless log_file_name.kind_of?(Pathname)
-      progname      = progname_from_file(log_file_name)
+    private_class_method def self.create_wrapper_logger(log_file, logger_class, wrapped_logger)
+      log_file = Pathname.new(log_file) if log_file.kind_of?(String)
+      log_file = ManageIQ.root.join("log", log_file) if log_file.try(:dirname).to_s == "."
+      progname = log_file.try(:basename, ".*").to_s
 
       logger_class.new(nil, :progname => progname).tap do |logger|
-        # HACK: In order to access the "filename" of the wrapped logger, we inject it as an instance var.
-        logger.instance_variable_set(:@log_file_name, log_file_name)
-
-        def logger.filename
```

Comment:
I thought filename was dropped in a follow-on to #21562, but apparently not?  Would it be possible to make this bit into a separate PR since it is really a tech debt thing that isn't related to this feature.

Reply:
This was added in commit e6ca841824fcbc27a8a260f391be17bb4506fc7e in this PR then removed in a5a51a10961934308c2c2e4307938f1c0822fc2b

I think the only way to "fix" this is to squash the commits.